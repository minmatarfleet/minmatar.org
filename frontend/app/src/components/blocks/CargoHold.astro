---
import type { CargoItem } from '@dtypes/layout_components'

interface Props {
    cargo:  CargoItem[];
    gap?:   string;
}

const {
    cargo,
    gap = 'var(--space-2xs)',
} = Astro.props

import Context from '@components/layout/Context.astro';
import Wrapper from '@components/compositions/Wrapper.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
import FlexInline from '@components/compositions/FlexInline.astro';
import ItemPicture from '@components/blocks/ItemPicture.astro';
import Badge from '@components/blocks/Badge.astro';
import CollapsableButton from '@components/blocks/CollapsableButton.astro';
---

<Flexblock
    gap='var(--space-3xs)'
    x-data={`{
        expanded: false,
        toggle_collapse() {
            this.expanded = !this.expanded
        }
    }`}
>
    <CollapsableButton x-on:click="toggle_collapse()">
        <FlexInline gap="var(--space-xs)">
            <slot />
        </FlexInline>
    </CollapsableButton>

    <Wrapper
        padding_block='0'
        padding_inline='var(--space-xl)'
    >
        <Context
            x-show="expanded"
            x-collapse.duration.0ms
            style="display: none"
        >
            {cargo.map( (item) =>
                <Badge
                    gap={gap}
                    size={32}
                    highlight={false}
                    title={`${item.amount} x ${item.name}`}
                >
                    <ItemPicture item_id={item.id} slot="picture" size={32} icon_quality={32} />
                </Badge>
            )}
        </Context>
    </Wrapper>
</Flexblock>