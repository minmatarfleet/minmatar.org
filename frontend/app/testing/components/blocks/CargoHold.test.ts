import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import CargoHold from "@components/blocks/CargoHold.astro";

test("CargoHold defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CargoHold, {
    props: {
      cargo: [
        {
          amont: 12,
          name: "chicken",
        },
      ],
    },
    slots: {
      default: "slot chickens",
    },
  });

  expect(result).toMatchSnapshot();
});
