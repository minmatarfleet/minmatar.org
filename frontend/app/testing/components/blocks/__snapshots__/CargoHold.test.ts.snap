// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CargoHold defaults 1`] = `
"<div class="flex-block gap---space-3xs" x-data="{
        expanded: false,
        toggle_collapse() {
            this.expanded = !this.expanded
        }
    }" data-mod style="--flex_block_gap: var(--space-3xs);">  <button x-on:click="toggle_collapse()" class="astro-muoml3se" type="button" data-astro-cid-muoml3se> <div class="fluid-fixed [ items-center ]" data-astro-cid-muoml3se="true" data-mod data-astro-cid-rwsocfon style="--fixed_fluid_gap: var(--space-l);--fixed_fluid_width: 20px;--fixed_fluid_breakpoint: 50%;">   <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: flex-start;">  slot chickens  </div>    <svg width="12" height="8" viewBox="0 0 12 8" fill="none" :class="expanded ? 'expanded' : 'collapsed'" data-astro-cid-muoml3se="true"> <path d="M10.59 0.000156279L6 4.58016L1.41 0.000155477L5.24537e-07 1.41016L6 7.41016L12 1.41016L10.59 0.000156279Z" fill="currentColor"></path> </svg>  </div>   </button>  <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-xl);--wrapper_padding_inline_end: var(--space-xl);--wrapper_padding_block: 0;--wrapper_max_width: 80rem;">  <div class="flex-block [ w-full ] gap---space-3xs" x-show="expanded" x-collapse.duration.0ms="true" style="display: none" data-mod style="--flex_block_gap: var(--space-3xs);">  <div class="fixed-fluid [ items-center ]" data-astro-cid-etghgjam="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-2xs);--fixed_fluid_width: 32px;--fixed_fluid_breakpoint: 50%;">  <picture class="animated" data-astro-cid-pp3yich5> <img loading="lazy" height="32" width="32" src="https://images.evetech.net/types/undefined/icon?size=32" alt="Item picture" data-astro-cid-pp3yich5> </picture>  <div class="flex-block gap-0" data-astro-cid-etghgjam="true" data-mod style="--flex_block_gap: 0;">  <span data-astro-cid-etghgjam="true" class="min-w-[200px]">chicken </span>     </div>   </div>     </div>   </div>    </div> "
`;
